korma/src/commonMain/kotlin/com/soywiz/korma/interpolation/Interpolation.kt

```diff
@@ -8,8 +8,12 @@ interface MutableInterpolable<T> {
     fun setToInterpolated(ratio: Double, l: T, r: T): T
 }
 
-fun Double.interpolate(l: Float, r: Float): Float = (l + (r - l) * this).toFloat()
+fun Float.interpolate(l: Float, r: Float): Float = (l + (r - l) * this)
```

Comment:
Can you add `Double`, `Int` and `Long` too? The receiver is the ratio, so would make sense to be able to interpolate primitives here